pyglmnet/utils.py

```diff
@@ -0,0 +1,66 @@
+import numpy as np
+
+
+def softmax(w):
+    """Softmax function of given array of number w
+
+    Parameters
+    ----------
+    w: array | list
+        The array of numbers.
+
+    Returns
+    -------
+    dist: array
+        The resulting array with values ranging from 0 to 1.
+    """
+    w = np.array(w)
+    maxes = np.amax(w, axis=1)
+    maxes = maxes.reshape(maxes.shape[0], 1)
+    e = np.exp(w - maxes)
+    dist = e / np.sum(e, axis=1, keepdims=True)
+    return dist
+
+
+def label_binarizer(y):
+    """Mimics scikit learn's LabelBinarizer
+
+
+    Parameters
+    ----------
+    y: ndarray (n_samples)
+        one dimensional array of class labels
+
+    Returns
+    -------
+    yb: array, shape (n_samples, n_classes)
+        one-hot encoding of labels in y
+    """
+    if y.ndim != 1:
+        raise ValueError('y has to be one-dimensional')
+    y_flat = y.ravel()
+    yb = np.zeros([len(y), y.max() + 1])
+    yb[np.arange(len(y)), y_flat] = 1
+    return yb
+
+
+def log_likelihood(y, yhat, distr):
```

Comment:
i would let the core `GLM` methods alone be part of the API. 

we created our own `softmax()` and `label_binarizer()` to avoid `scikit-learn` dependency.